File: LLM-Translations/GPT-4/codenet/C++/Go/s154250720.go

package main

import (
	"fmt"
	"sort"
)

const maxn = 200000 + 10
const INF = int(1e8)

var maps [maxn]int
var num [maxn]int

func main() {
	var n int
	for {
		_, err := fmt.Scanf("%d", &n)
		if err != nil {
			break
		}
		for i := 1; i <= n; i++ {
			fmt.Scanf("%d", &maps[i])
			num[i] = maps[i]
		}
		sort.Slice(maps[1:n+1], func(i, j int) bool {
			return maps[i+1] < maps[j+1]
		})
		m := n / 2
		for i := 1; i <= n; i++ {
			if num[i] <= maps[m] {
				fmt.Printf("%d\n", maps[m+1])
			} else {
				fmt.Printf("%d\n", maps[m])
			}
		}
	}
}
// 

File: LLM-Translations/GPT-4/codenet/C++/C/s154250720.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
#include<math.h>
#include<limits.h>

int compare(const void *a, const void *b) {
    return (*(int *)a - *(int *)b);
}

int main() {
    int n;
    while(scanf("%d",&n) != EOF) {
        int maps[n+1];
        int num[n+1];
        for(int i=1; i<=n; i++) {
            scanf("%d",&maps[i]);
            num[i]=maps[i];
        }
        qsort(maps+1, n, sizeof(int), compare);
        int m=n/2;
        for(int i=1; i<=n; i++) {
            if(num[i]<=maps[m])
                printf("%d\n",maps[m+1]);
            else
                printf("%d\n",maps[m]);

        }
    }
    return 0;
}
// 
